src/source.rs

```diff
@@ -146,3 +195,137 @@ pub fn source_remove(source_id: Id) {
         glib_ffi::g_source_remove(source_id.to_glib());
     }
 }
+
+/// The priority of sources
+///
+#[derive(Clone, Copy, Debug, Default, Eq, PartialEq)]
+pub struct Priority(i32);
+
+impl ToGlib for Priority {
+    type GlibType = i32;
+
+    #[inline]
+    fn to_glib(&self) -> i32 {
+        self.0
+    }
+}
+
+impl FromGlib<i32> for Priority {
+    #[inline]
+    fn from_glib(val: i32) -> Priority {
+        Priority(val)
+    }
+}
+
+pub const PRIORITY_HIGH: Priority = Priority(glib_ffi::G_PRIORITY_HIGH);
+pub const PRIORITY_DEFAULT: Priority = Priority(glib_ffi::G_PRIORITY_DEFAULT);
+pub const PRIORITY_HIGH_IDLE: Priority = Priority(glib_ffi::G_PRIORITY_HIGH_IDLE);
+pub const PRIORITY_DEFAULT_IDLE: Priority = Priority(glib_ffi::G_PRIORITY_DEFAULT_IDLE);
+pub const PRIORITY_LOW: Priority = Priority(glib_ffi::G_PRIORITY_LOW);
+
+/// Adds a closure to be called by the main loop the return `Source` is attached to when it's idle.
+///
+/// `func` will be called repeatedly until it returns `Continue(false)`.
+pub fn idle_source_new<'a, N: Into<Option<&'a str>>, F>(name: N, priority: Priority, func: F) -> Source
+where F: FnMut() -> Continue + Send + 'static {
```

Comment:
The question here is mostly whether we want single functions for creating Sources, or create something with a builder pattern here. Single functions seems more simple to me